server/sonar-server/src/main/java/org/sonar/server/measure/custom/ws/CreateAction.java

```diff
@@ -0,0 +1,291 @@
+        .setCreatedAt(now);
+      setMeasureValue(measure, request, metric);
+      dbClient.customMeasureDao().insert(dbSession, measure);
+      dbSession.commit();
+
+      JsonWriter json = response.newJsonWriter();
+      writeMeasure(json, measure, component, metric, request.mandatoryParam(PARAM_VALUE));
+      json.close();
+    } finally {
+      MyBatis.closeQuietly(dbSession);
+    }
+  }
+
+  private void checkPermissions(ComponentDto component) {
+    if (userSession.hasGlobalPermission(GlobalPermissions.SYSTEM_ADMIN)) {
+      return;
+    }
+
+    userSession.checkLoggedIn().checkProjectUuidPermission(UserRole.ADMIN, component.projectUuid());
+  }
+
+  private void checkMeasureDoesNotExistAlready(DbSession dbSession, ComponentDto component, MetricDto metric) {
+    int nbMeasuresOnSameMetricAndMeasure = dbClient.customMeasureDao().countByComponentIdAndMetricId(dbSession, component.uuid(), metric.getId());
+    if (nbMeasuresOnSameMetricAndMeasure > 0) {
+      throw new ServerException(HttpURLConnection.HTTP_CONFLICT, String.format("A measure already exists for project id '%s' and metric id '%d'", component.uuid(), metric.getId()));
+    }
+  }
+
+  private void writeMeasure(JsonWriter json, CustomMeasureDto measure, ComponentDto component, MetricDto metric, String measureWithoutInternalFormatting) {
+    json.beginObject();
+    json.prop(FIELD_ID, String.valueOf(measure.getId()));
+    json.prop(FIELD_METRIC_ID, String.valueOf(metric.getId()));
+    json.prop(FIELD_METRIC_KEY, metric.getKey());
+    json.prop(FIELD_PROJECT_ID, component.uuid());
+    json.prop(FIELD_PROJECT_KEY, component.key());
+    json.prop(FIELD_DESCRIPTION, measure.getDescription());
+    json.prop(FIELD_VALUE, measureWithoutInternalFormatting);
+    json.endObject();
+  }
+
+  private void setMeasureValue(CustomMeasureDto measure, Request request, MetricDto metric) {
+    String valueAsString = request.mandatoryParam(PARAM_VALUE);
+    Metric.ValueType metricType = Metric.ValueType.valueOf(metric.getValueType());
+    try {
+      switch (metricType) {
+        case BOOL:
+          typeValidations.validate(valueAsString, PropertyType.BOOLEAN.name(), null);
+          measure.setValue(Boolean.parseBoolean(valueAsString) ? 1.0d : 0.0d);
+          break;
+        case INT:
+        case MILLISEC:
+          typeValidations.validate(valueAsString, PropertyType.INTEGER.name(), null);
+          measure.setValue(Integer.parseInt(valueAsString));
+          break;
+        case WORK_DUR:
+          typeValidations.validate(valueAsString, PropertyType.METRIC_WORK_DURATION.name(), null);
+          measure.setValue(durations.decode(valueAsString).toMinutes());
+          break;
+        case FLOAT:
+        case PERCENT:
+        case RATING:
+          typeValidations.validate(valueAsString, PropertyType.FLOAT.name(), null);
+          measure.setValue(Double.parseDouble(valueAsString));
+          break;
+        case LEVEL:
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) End this switch case with an unconditional break, continue, return or throw statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS128)